platforms/documentation/docs/src/docs/userguide/authoring-builds/other/validation_problems.adoc

```diff
@@ -433,9 +433,9 @@ To address this, we suggest to use `java.net.URI` instead.
 
 This error indicates that a nested map declares a key of an unsupported type.
 Gradle uses the key to generate a name for the (sub)property.
-Only allowing keys of type `Enum`, `Integer`, and `String` guarantees that these names are unique and well-formed, and is preferable to relying on `toString()` for producing such names.
+Only allowing keys of certain types guarantees that these names are unique and well-formed. This is preferable to relying on `toString()` for producing such names.
 
-To fix this problem, change the type of the key to `Enum`, `Integer`, or `String`.
+To fix this problem, change the type of the key to `Integer` or `String` or an enum.
```

Comment:
🤔 I think `Integer` is the least good candidate for a key, so I wouldn't start with that. The best is probably an enum, as it's the most descriptive. So I think it would make sense to propose them in that order: enum, `String` or `Integer`. WDYT?